main/src/visualization/plotPerf.js

```diff
@@ -0,0 +1,137 @@
+const D3Node = require('d3-node');
+const fs = require('fs');
+const d3 = require('d3');
+const logsReader = require('../filesIO/logsReader');
+const styles = `
+.bar rect {
+  fill: steelblue;
+}
+.bar text {
+  fill: #fff;
+  font: 10px sans-serif;
+}
+.drag rect {
+  fill: orange
+}
+.bar text {
+  fill: #fff;
+  font: 10px sans-serif;
+}`;
+
+const options = {
+  svgStyles: styles,
+  d3Module: d3,
+};
+
+function SVGFromLogs(path) {
+
+  const d3n = new D3Node(options);
+
+  logsReader.initLogs('../../out/mapbox/perfLogs.json');
+  const frameTimes = logsReader.getFrameTimes();
+  const instantFPS = logsReader.getInstantFPS();
+  const timeBetweenFrames = logsReader.getTimeBetweenFrames();
+  const dragEvents = logsReader.getStartAndEndOfDragEvents();
+  const DRAG_EVENTS_HEIGHT = 30;
+
+  const margin = {
+    top: 10, right: 30, bottom: 30, left: 30,
+  };
+  const labelMargin = {
+    left: 30, bottom: 30,
+  };
+  const width = 960 - labelMargin.left - margin.left - margin.right;
+  const height = 500 - labelMargin.bottom - margin.top - margin.bottom;
+
+  const maxFps = d3.max(instantFPS);
+  const xOtherScale = d3.scaleLinear()
+    .domain([0, frameTimes[frameTimes.length - 1] - frameTimes[0]])
+    .range([0, width]);
+
+  const x = d3.scaleLinear()
+    .domain([frameTimes[0] - timeBetweenFrames[0], frameTimes[frameTimes.length - 1]])
+    .range([0, width]);
+
+  const y = d3.scaleLinear()
+    .domain([0, d3.max(instantFPS)])
+    .range([height, 0]);
+
+
+  const svgWidth = width + labelMargin.left + margin.left + margin.right;
+  const svgHeight = height + labelMargin.bottom + margin.top + margin.bottom;
+
+  function getXOffset(data, index) {
+    return x(data - timeBetweenFrames[index]);
+  }
+
+  function getYOffset(index) {
+    return y(instantFPS[index]);
+  }
+
+  const svg = d3n.createSVG(svgWidth, svgHeight)
+    .append('g')
+    .attr('transform', `translate(${margin.left},${margin.top})`);
+
+  // Label Names
+  // text label for the x axis
```

Comment:
Note for a possible follow-up PR:

Often the fact that you feel the need to add a comment is a good indication that the code below it could go into a separate function. E.g. here you could add 3 functions:
```
function addLabelNames() {
  addXAxisLabels();
  addYAxisLabels();
}
function addXAxisLabels() {
  // ...
}
function addYAxisLabels() {
  // ...
}
```
Since you need access to some other variables, you could also turn the module into a class, where `width`, `height`, etc are class variables.